provisioning_client/adapters/hsm_client_riot.c

```diff
@@ -275,13 +264,13 @@ static int produce_device_cert(HSM_CLIENT_X509_INFO* riot_info, RIOT_ECC_SIGNATU
     {
         // Generating "root"-signed DeviceID certificate
         DERInitContext(&der_ctx, der_buffer, DER_MAX_TBS);
-        if (X509GetDeviceCertTBS(&der_ctx, &X509_DEVICE_TBS_DATA, &riot_info->device_id_pub) != 0)
+        if (X509GetDeviceCertTBS(&der_ctx, &X509_DEVICE_TBS_DATA, &riot_info->device_id_pub, (uint8_t*)&riot_info->device_id_pub, sizeof(riot_info->device_id_pub)) != 0)
```

Comment:
Same as above.

Reply:
same as above

Reply:
This one is a problem.  Since the DeviceID cert is signed but the root CA key in this case, the Authority Key Identifier does not correspond to the DeviceID key.  It is the root CA key.

Reply:
Fixed